### DISCUSSION TITLE
createdAt, updatedAt are coming out as invalid date

### DISCUSSION THREAD

**Discussion by thousynth** [sequelize/sequelize#16482] [Help & Questions]
I created a sqlite database and seeded it with some data.
Table after the process is filled and looks like below:

id | quiz_name | created_at | updated_at | createdAt | updatedAt
-- | -- | -- | -- | -- | --
1 | General Knowledge Quiz 1 | 2023-09-11 17:09:19.973 +00:00 | 2023-09-11 17:09:19.973 +00:00 | 2023-09-11 17:09:19.972 +00:00 | 2023-09-11 17:09:19.972 +00:00
2 | Math Quiz 1 | 2023-09-11 17:09:19.990 +00:00 | 2023-09-11 17:09:19.990 +00:00 | 2023-09-11 17:09:19.989 +00:00 | 2023-09-11 17:09:19.989 +00:00
3 | Science Quiz 1 | 2023-09-11 17:09:19.999 +00:00 | 2023-09-11 17:09:19.999 +00:00 | 2023-09-11 17:09:19.999 +00:00 | 2023-09-11 17:09:19.999 +00:00

The issue is the `createdAt` and `updateAt` field is coming out as invalid date when I'm trying to query this via an endpoint:

Hono endpoint:
```js
app.get("/api/quiz", async (c) => {
  try {
    const quizzes = await Quiz.findByPk(1, {
      include: [
        {
          model: Question,
          include: [Option],
        },
      ],
    });

    console.log(quizzes?.toJSON());
    return c.json(quizzes);
  } catch (error) {
    console.error(error);
  }
});
```

The output at console:
```js
{
  id: 1,
  quiz_name: "General Knowledge Quiz 1",
  createdAt: Invalid Date,
  updatedAt: Invalid Date,
  Questions: [
    {
      id: 1,
      question_text: "What is the capital of France?",
      createdAt: Invalid Date,
      updatedAt: Invalid Date,
      QuizId: 1,
      Options: [
        {
          id: 1,
          option_text: "London",
          is_correct: false,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 1
        }, {
          id: 3,
          option_text: "Madrid",
          is_correct: false,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 1
        }, {
          id: 2,
          option_text: "Paris",
          is_correct: true,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 1
        }
      ]
    }, {
      id: 2,
      question_text: "Who wrote the play \"Romeo and Juliet\"?",
      createdAt: Invalid Date,
      updatedAt: Invalid Date,
      QuizId: 1,
      Options: [
        {
          id: 5,
          option_text: "Charles Dickens",
          is_correct: false,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 2
        }, {
          id: 6,
          option_text: "Jane Austen",
          is_correct: false,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 2
        }, {
          id: 4,
          option_text: "William Shakespeare",
          is_correct: true,
          createdAt: Invalid Date,
          updatedAt: Invalid Date,
          QuestionId: 2
        }
      ]
    }
  ]
}
```

I've tried adding a custom createdAt, and updatedAt field using `beforeCreate` and `beforeUpdate` hooks on the model but it didn't work. The values were still `Invalid Date`.

<details>
<summary>`Quiz Model`</summary>

```js
  const Quiz = db.define("Quiz", {
    quiz_name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
  });
```
</details>

**Comment by ephys** [comment]
There should be no space before the timezone offset in the ISO 8601 date format. Your dates should look like this:

> 2023-09-11 17:09:19.973+00:00

instead of this:

> 2023-09-11 17:09:19.973 +00:00

**Reply by thousynth** [reply]
but these columns are filled automatically at the time of insertion. How do I ensure that the space isn't there?

**Reply by ephys** [reply]
Do you mean you did not include them in your seed, or specify them yourself and that Sequelize inserted them?

In that case, I recommend opening a bug report with a reproducible example so we can fix the issue
